modules/apps/27-interchain-accounts/controller/keeper/handshake.go

```diff
@@ -58,15 +67,15 @@ func (k Keeper) OnChanOpenInit(
 		}
 
 		if channel.State == channeltypes.OPEN {
-			return sdkerrors.Wrapf(icatypes.ErrActiveChannelAlreadySet, "existing active channel %s for portID %s is already OPEN", activeChannelID, portID)
+			return "", sdkerrors.Wrapf(icatypes.ErrActiveChannelAlreadySet, "existing active channel %s for portID %s is already OPEN", activeChannelID, portID)
 		}
 
 		if !icatypes.IsPreviousMetadataEqual(channel.Version, metadata) {
-			return sdkerrors.Wrap(icatypes.ErrInvalidVersion, "previous active channel metadata does not match provided version")
+			return "", sdkerrors.Wrap(icatypes.ErrInvalidVersion, "previous active channel metadata does not match provided version")
 		}
 	}
 
-	return nil
+	return string(icatypes.ModuleCdc.MustMarshalJSON(&metadata)), nil
```

Comment:
just to make sure I understand, if this panics this is handled at the sdk level right?

Reply:
Yep, its caught by the grpc recovery interceptor (essentially a middleware handler in grpc land)